### PR TITLE
Add note to address fix of Dex/Gangway replica off-balanced issue (bsc#1157337)

### PR DESCRIPTION
**Pull request by c3y1huang**
# Make sure it's tested
### Status **BEFORE** applying the patch
```
oidc-dex-5566f77f89-4g5w2               1/1     Running   0       25m     10.244.0.155   caasp-master-cyh-cluster-0   <none>           <none>
oidc-dex-5566f77f89-gxqrk               1/1     Running   0       25m     10.244.0.205   caasp-master-cyh-cluster-0   <none>           <none>
oidc-dex-5566f77f89-tq7p9               1/1     Running   1       25m     10.244.0.189   caasp-master-cyh-cluster-0   <none>           <none>
oidc-gangway-79447879fb-772gj           1/1     Running   0       25m     10.244.0.47    caasp-master-cyh-cluster-0   <none>           <none>
oidc-gangway-79447879fb-j8dmt           1/1     Running   0       25m     10.244.0.146   caasp-master-cyh-cluster-0   <none>           <none>
oidc-gangway-79447879fb-mqphq           1/1     Running   0       25m     10.244.0.195   caasp-master-cyh-cluster-0   <none>           <none>
```

#### Skuba Node Join
```
---1*MASTER
oidc-dex-5566f77f89-jcss4               1/1     Running   0       3m23s   10.244.0.117   caasp-master-c3y1-cluster-0   <none>           <none>
oidc-dex-5566f77f89-l2bxt               1/1     Running   1       3m23s   10.244.0.251   caasp-master-c3y1-cluster-0   <none>           <none>
oidc-gangway-79447879fb-49wmz           1/1     Running   0       3m45s   10.244.0.145   caasp-master-c3y1-cluster-0   <none>           <none>
oidc-gangway-79447879fb-frjxh           1/1     Running   0       3m45s   10.244.0.9     caasp-master-c3y1-cluster-0   <none>           <none>
oidc-gangway-79447879fb-wwk48           1/1     Running   0       3m45s   10.244.0.224   caasp-master-c3y1-cluster-0   <none>           <none>
---2*MASTER
oidc-dex-768b4f4778-f479g               1/1     Running   0       97s     10.12.148.16   caasp-master-c3y1-cluster-0   <none>           <none>
oidc-dex-768b4f4778-n7q9r               1/1     Running   0       98s     10.244.1.59    caasp-master-c3y1-cluster-1   <none>           <none>
oidc-gangway-788b4c5696-4r6mc           1/1     Running   0       98s     10.12.3.79     caasp-master-c3y1-cluster-0   <none>           <none>
oidc-gangway-788b4c5696-s99mk           1/1     Running   0       21s     10.244.1.199   caasp-master-c3y1-cluster-1   <none>           <none>
oidc-gangway-788b4c5696-x9bfv           1/1     Running   0       98s     10.244.1.101   caasp-master-c3y1-cluster-1   <none>           <none>
---2*MASTER,1*WORKER
oidc-dex-768b4f4778-f479g               1/1     Running   0       11m     10.12.148.16   caasp-master-c3y1-cluster-0   <none>           <none>
oidc-dex-768b4f4778-n7q9r               1/1     Running   0       11m     10.244.1.59    caasp-master-c3y1-cluster-1   <none>           <none>
oidc-gangway-d776d65dd-9v6d6            1/1     Running   0       6m25s   10.244.2.135   caasp-worker-c3y1-cluster-1   <none>           <none>
oidc-gangway-d776d65dd-cxnn6            1/1     Running   0       6m25s   10.12.74.227   caasp-master-c3y1-cluster-0   <none>           <none>
oidc-gangway-d776d65dd-mf5jc            1/1     Running   0       32s     10.244.1.132   caasp-master-c3y1-cluster-1   <none>           <none>
---2*MASTER,2*WORKER
oidc-dex-768b4f4778-f479g               1/1     Running   0       12m     10.12.148.16   caasp-master-c3y1-cluster-0   <none>           <none>
oidc-dex-768b4f4778-n7q9r               1/1     Running   0       12m     10.244.1.59    caasp-master-c3y1-cluster-1   <none>           <none>
oidc-gangway-d776d65dd-9v6d6            1/1     Running   0       7m56s   10.244.2.135   caasp-worker-c3y1-cluster-1   <none>           <none>
oidc-gangway-d776d65dd-cxnn6            1/1     Running   0       7m56s   10.12.74.227   caasp-master-c3y1-cluster-0   <none>           <none>
oidc-gangway-d776d65dd-mf5jc            1/1     Running   0       2m3s    10.244.1.132   caasp-master-c3y1-cluster-1   <none>           <none>
---3*MASTER,2*WORKER
oidc-dex-768b4f4778-f479g               1/1     Running   0       14m     10.12.148.16   caasp-master-c3y1-cluster-0   <none>           <none>
oidc-dex-768b4f4778-n7q9r               1/1     Running   0       14m     10.244.1.59    caasp-master-c3y1-cluster-1   <none>           <none>
oidc-gangway-d776d65dd-9v6d6            1/1     Running   0       9m40s   10.244.2.135   caasp-worker-c3y1-cluster-1   <none>           <none>
oidc-gangway-d776d65dd-cxnn6            1/1     Running   0       9m40s   10.12.74.227   caasp-master-c3y1-cluster-0   <none>           <none>
oidc-gangway-d776d65dd-mf5jc            1/1     Running   0       3m47s   10.244.1.132   caasp-master-c3y1-cluster-1   <none>           <none>
```
#### Skuba Node Remove
```
---3*MASTER,2*WORKER
oidc-dex-768b4f4778-f479g               1/1     Running   0       14m     10.12.148.16   caasp-master-c3y1-cluster-0   <none>           <none>
oidc-dex-768b4f4778-n7q9r               1/1     Running   0       14m     10.244.1.59    caasp-master-c3y1-cluster-1   <none>           <none>
oidc-gangway-d776d65dd-9v6d6            1/1     Running   0       9m40s   10.244.2.135   caasp-worker-c3y1-cluster-1   <none>           <none>
oidc-gangway-d776d65dd-cxnn6            1/1     Running   0       9m40s   10.12.74.227   caasp-master-c3y1-cluster-0   <none>           <none>
oidc-gangway-d776d65dd-mf5jc            1/1     Running   0       3m47s   10.244.1.132   caasp-master-c3y1-cluster-1   <none>           <none>
---2*MASTER,2*WORKER
oidc-dex-768b4f4778-f479g               1/1     Running   0       17m     10.12.148.16   caasp-master-c3y1-cluster-0   <none>           <none>
oidc-dex-768b4f4778-n7q9r               1/1     Running   0       18m     10.244.1.59    caasp-master-c3y1-cluster-1   <none>           <none>
oidc-gangway-d776d65dd-9v6d6            1/1     Running   0       13m     10.244.2.135   caasp-worker-c3y1-cluster-1   <none>           <none>
oidc-gangway-d776d65dd-cxnn6            1/1     Running   0       13m     10.12.74.227   caasp-master-c3y1-cluster-0   <none>           <none>
oidc-gangway-d776d65dd-mf5jc            1/1     Running   0       7m16s   10.244.1.132   caasp-master-c3y1-cluster-1   <none>           <none>
---2*MASTER,1*WORKER
oidc-dex-768b4f4778-f479g               1/1     Running   0       19m     10.12.148.16   caasp-master-c3y1-cluster-0   <none>           <none> 
oidc-dex-768b4f4778-n7q9r               1/1     Running   0       19m     10.244.1.59    caasp-master-c3y1-cluster-1   <none>           <none>                                   
oidc-gangway-d776d65dd-9v6d6            1/1     Running   0       14m     10.244.2.135   caasp-worker-c3y1-cluster-1   <none>           <none>                                   
oidc-gangway-d776d65dd-cxnn6            1/1     Running   0       14m     10.12.74.227   caasp-master-c3y1-cluster-0   <none>           <none>                                   
oidc-gangway-d776d65dd-mf5jc            1/1     Running   0       8m17s   10.244.1.132   caasp-master-c3y1-cluster-1   <none>           <none>
---2*MASTER
oidc-dex-768b4f4778-f479g               1/1     Running   0       22m     10.12.148.16    caasp-master-c3y1-cluster-0   <none>           <none>
oidc-dex-768b4f4778-n7q9r               1/1     Running   0       22m     10.244.1.59     caasp-master-c3y1-cluster-1   <none>           <none>
oidc-gangway-788b4c5696-5jm76           1/1     Running   0       78s     10.12.96.60     caasp-master-c3y1-cluster-0   <none>           <none>
oidc-gangway-788b4c5696-9cp55           1/1     Running   0       52s     10.12.85.175    caasp-master-c3y1-cluster-0   <none>           <none>
oidc-gangway-788b4c5696-b7bqb           1/1     Running   0       77s     10.12.133.100   caasp-master-c3y1-cluster-0   <none>           <none>
---1*MASTER
oidc-dex-7755744d84-kp9l7               1/1     Running   0       2m9s    10.12.45.54     caasp-master-c3y1-cluster-0   <none>           <none>
oidc-dex-7755744d84-wb8s9               1/1     Running   0       2m8s    10.12.197.137   caasp-master-c3y1-cluster-0   <none>           <none>
oidc-gangway-788b4c5696-5jm76           1/1     Running   0       4m28s   10.12.96.60     caasp-master-c3y1-cluster-0   <none>           <none>
oidc-gangway-788b4c5696-qrdt8           1/1     Running   0       2m5s    10.12.130.120   caasp-master-c3y1-cluster-0   <none>           <none>
oidc-gangway-788b4c5696-v5x2f           1/1     Running   0       2m4s    10.12.175.212   caasp-master-c3y1-cluster-0   <none>           <none>
```

*Technical writers will not always be able to verify the implementation!*

Any PR opened is assumed to have been verified by QA if not designated by comments or labels otherwise. Please make sure you have tested the changes and included any information that a user would require to use the documentation.

# Describe your changes
Add podAntiAffinity to avoid Dex and Gangway being scheduled
onto same node.
Rfc: https://github.com/SUSE/caasp-rfc/pull/50

# Related Issues / Projects
Please provide links to Bugzilla and other GitHub projects with your description if they are related to the changes.

Relates to:  
* [#1157337](https://bugzilla.suse.com/show_bug.cgi?id=1157337)
* https://github.com/SUSE/skuba/pull/860
* https://github.com/SUSE/avant-garde/issues/1093
* https://github.com/SUSE/doc-caasp/issues/632

# Enable maintainer updates
Please enable maintainer updates so we can push commits into your branch to make collaboration and reviews easier.

# Do not force push your branch
Please avoid force pushing to branches that are subject of pull requests. Force pushing breaks maintainer commits in many cases and is very hard (if not impossible) to untangle for backporting.

# Labels
Please set any (and all) appropriate labels that describe the status of the PR.

| **Label** | **Description** |
| --- | --- |
| P1 | PR should be worked on and merged as soon as possible |
| Blocked | Work can not proceed because other work has not been completed, PR can not be merged (code has not been merged but documentation is ready) |
| On-Hold | Underlying work is completed but the PR should not be merged |
| ReleaseNotes | User interaction is required after the introduction of this change and the change must be mentioned in the release notes |
| v3/v4/v4.x | Which version of the release the PR should be merged into, this can be multiple versions, please set the "Backport" label if it needs to go into a previous release |
| Needs Review | Some details of the PR are known to be incomplete and must be discussed with other engineers before merging (if possible assign reviewers or cc mention in comments), PR can not be merged |

### CODE REVIEW

File: adoc/deployment-sysreqs.adoc [hunk 1]
```diff
@@ -27,6 +27,9 @@ The minimal viable failure tolerant production environment configuration consist
 * 3 master nodes
 * 2 worker nodes
 
+When cluster size not reaching desired number, the replica pods such as `oidc-dex` and `oidc-gangway`
```
Comment (r0ckarong) [inline adoc/deployment-sysreqs.adoc]: What does "desired number" mean here? Do I understand this correctly that the minimum configuration is not large enough to hold these replica pods?

**Comment by ereslibre** [comment]
This is blocked until https://github.com/SUSE/skuba/pull/860 merges. It goes for a solution that requires this documentation, but another solution is feasible a well that wouldn't require documentation at all.

**Comment by c3y1huang** [comment]
@r0ckarong , [fix](https://github.com/SUSE/skuba/pull/860) merged. Please help with review and merge this.

**Comment by maximenoel8** [comment]
See my comment : https://github.com/SUSE/skuba/pull/860

**Pull request merged**